### PR TITLE
fix: save elevation on site creation

### PR DESCRIPTION
**Pull request by paulschreiber**
## Description
Save elevation to database on site creation.

### Related Issues
Addresses https://github.com/techmatters/terraso-product/issues/774
Depends on https://github.com/techmatters/terraso-mobile-client/pull/1315

### CODE REVIEW

File: dev-client/src/screens/HomeScreen/components/TemporaryLocationCallout.tsx [hunk 1]
```diff
@@ -52,16 +52,23 @@ export const TemporaryLocationCallout = ({
   const {t} = useTranslation();
   const navigation = useNavigation();
   const [siteElevationString, setSiteElevationString] = useState('');
+  const [siteElevationValue, setSiteElevationValue] = useState(0);
```
Comment (knipec) [inline dev-client/src/screens/HomeScreen/components/TemporaryLocationCallout.tsx:55]: Suggestion: Can we remove the siteElevationString state stuff now? Like, just do `useState` for the elevation value, and then derive the elevation string from the value when we need the string? It seems less clean to save both as separate state variables when one is easily derived from the other.

File: dev-client/src/screens/LocationScreens/components/CreateSiteButton.tsx [hunk 2]
```diff
@@ -25,15 +25,15 @@ import {Coords} from 'terraso-client-shared/types';
 import {Icon} from 'terraso-mobile-client/components/icons/Icon';
 import {useNavigation} from 'terraso-mobile-client/navigation/hooks/useNavigation';
 
-type Props = {coords: Coords};
+type Props = {coords: Coords; elevation?: number};
```
Comment (knipec) [inline dev-client/src/screens/LocationScreens/components/CreateSiteButton.tsx:28]: Question: Why is `elevation` optional -- is there ever a case where we wouldn't have a value for `elevation` here?

**Review by knipec — APPROVED** [review]
Looks good to me after addressing the two comments!

File: dev-client/src/screens/CreateSiteScreen/components/CreateSiteView.tsx [hunk 3]
```diff
@@ -66,17 +64,14 @@ export const CreateSiteView = ({
 
   const onSave = useCallback(
     async ({...form}: FormState) => {
-      const {coords, ...site} = validationSchema.cast(form);
-      const createdSite = await createSiteCallback({
-        ...site,
-        ...parseCoords(coords),
-      });
+      const {...site} = validationSchema.cast(form);
+      const createdSite = await createSiteCallback({...site, elevation});
```
Comment (knipec) [inline dev-client/src/screens/CreateSiteScreen/components/CreateSiteView.tsx:68]: You said you plan for a future PR to handle the case where the user has changed the lat/long in the form, right?

File: dev-client/src/screens/LocationScreens/components/CreateSiteButton.tsx (hunk 2 above)
Comment (paulschreiber) [reply dev-client/src/screens/LocationScreens/components/CreateSiteButton.tsx:28]: It's possible the elevation API could fail.

File: dev-client/src/screens/CreateSiteScreen/components/CreateSiteView.tsx (hunk 3 above)
Comment (paulschreiber) [reply dev-client/src/screens/CreateSiteScreen/components/CreateSiteView.tsx:68]: Yes.

File: dev-client/src/screens/HomeScreen/components/TemporaryLocationCallout.tsx (hunk 1 above)
Comment (paulschreiber) [reply dev-client/src/screens/HomeScreen/components/TemporaryLocationCallout.tsx:56]: Good catch.

**Pull request merged**